### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by LesediMoko**
This pull request includes project setup code such as installations and file+folder creations. 

Along with that, I have also setup my types and store for my state management (zustand). 

The Tanstack Query code and the Trending component can be ignored as I was just messing around to see if it would work. They do not form part of this PR 😅

### CODE REVIEW

File: src/services/api.service.ts
Comment (yanevdw) [inline src/services/api.service.ts:1]: I believe the naming convention for TypeScript files should be kebab-case, i.e. `api-service.ts`

**Review by yanevdw — APPROVED** [review]
Great work! Just check the comment on the naming convention of `api.service.ts`.

**Review by Hugovs2000 — APPROVED** [review]
Quite a cool setup you have with the api calls, I see no problems 👍🏻 although `api-service.ts` would be a better name as Yane mentioned

File: src/services/api.service.ts
Comment (LesediMoko) [reply src/services/api.service.ts:1]: You're right, change has been made. Thank you!

**Pull request merged**